<!-- Runs obstacle_detector with a static_scan publisher. -->
<launch>
  <node pkg="obstacle_detector" type="static_scan_publisher" name="static_scan_publisher" output="screen"/>
  
  <node pkg="tf" type="static_transform_publisher" name="world_to_scanner" output="screen" args="0 0 0 0 0 0 world scanner 10"/>
 
  <node pkg="obstacle_detector" type="obstacle_detector" name="obstacle_detector" output="screen">
    <param name="world_frame"          value="world"/>
    <param name="scanner_frame"        value="scanner"/>
 
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>
    <param name="transform_to_world"   value="true"/>
    <param name="use_split_and_merge"  value="false"/>

    <param name="min_group_points"     value="3"/>
    <param name="max_group_distance"   value="0.100"/>
    <param name="distance_proportion"  value="0.006136"/>
    <param name="max_split_distance"   value="0.100"/>

    <param name="max_merge_separation" value="0.200"/>
    <param name="max_merge_spread"     value="0.070"/>
    <param name="max_circle_radius"    value="0.500"/>
    <param name="radius_enlargement"   value="0.050"/>

    <param name="max_scanner_range"    value="6.0"/>
    <param name="max_x_range"          value="5.0"/>
    <param name="min_x_range"          value="-5.0"/>
    <param name="max_y_range"          value="5.0"/>
    <param name="min_y_range"          value="-5.0"/>
  </node>
  
  <node pkg="obstacle_detector" type="obstacle_tracker" name="obstacle_tracker" output="screen">
    <param name="fade_counter"            value="50"/>
    <param name="pose_measure_variance"   value="1.0"/>
    <param name="radius_measure_variance" value="1.0"/>
    <param name="pose_process_variance"   value="0.2"/>    
    <param name="radius_process_variance" value="0.2"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_visualizer" name="obstacle_visualizer" output="screen">
    <param name="circles_color"  value="3"/>
    <param name="segments_color" value="2"/>
    <param name="alpha"          value="0.7"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_visualizer" name="tracked_obstacle_visualizer" output="screen">
    <param name="circles_color"  value="4"/>
    <param name="segments_color" value="2"/>
    <param name="alpha"          value="0.7"/>

    <remap from="obstacles" to="tracked_obstacles"/>
    <remap from="obstacles_markers" to="tracked_obstacles_markers"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find obstacle_detector)/resources/obstacle_detector.rviz"/>
</launch>

